packages/prompts/src/autocomplete.ts

```diff
@@ -0,0 +1,298 @@
+								)
+							: '';
+
+					// Render options with selection
+					const displayOptions =
+						this.filteredOptions.length === 0
+							? []
+							: limitOptions({
+									cursor: this.cursor,
+									options: this.filteredOptions,
+									style: (option, active) => {
+										const label = getLabel(option);
+										const hint =
+											option.hint && option.value === this.focusedValue
+												? color.dim(` (${option.hint})`)
+												: '';
+
+										return active
+											? `${color.green(S_RADIO_ACTIVE)} ${label}${hint}`
+											: `${color.dim(S_RADIO_INACTIVE)} ${color.dim(label)}${hint}`;
+									},
+									maxItems: opts.maxItems,
+									output: opts.output,
+								});
+
+					// Show instructions
+					const instructions = [
+						`${color.dim('↑/↓')} to select`,
+						`${color.dim('Enter:')} confirm`,
+						`${color.dim('Type:')} to search`,
+					];
+
+					// No matches message
+					const noResults =
+						this.filteredOptions.length === 0 && valueAsString
+							? [`${color.cyan(S_BAR)}  ${color.yellow('No matches found')}`]
+							: [];
+
+					// Return the formatted prompt
+					return [
+						title,
+						`${color.cyan(S_BAR)}  ${color.dim('Search:')} ${searchText}${matches}`,
+						...noResults,
+						...displayOptions.map((option) => `${color.cyan(S_BAR)}  ${option}`),
+						`${color.cyan(S_BAR)}  ${color.dim(instructions.join(' • '))}`,
+						`${color.cyan(S_BAR_END)}`,
+					].join('\n');
+				}
+			}
+		},
+	});
+
+	// Return the result or cancel symbol
+	return prompt.prompt() as Promise<Value | symbol>;
+};
+
+// Type definition for the autocompleteMultiselect component
+export interface AutocompleteMultiSelectOptions<Value> {
+	/**
+	 * The message to display to the user
+	 */
+	message: string;
+	/**
+	 * The options for the user to choose from
+	 */
+	options: Option<Value>[];
+	/**
+	 * The initial selected values
+	 */
+	initialValues?: Value[];
+	/**
+	 * The maximum number of items that can be selected
+	 */
+	maxItems?: number;
+	/**
+	 * The placeholder to display in the input
+	 */
+	placeholder?: string;
+	/**
+	 * The stream to read from
+	 */
+	input?: NodeJS.ReadStream;
+	/**
+	 * The stream to write to
+	 */
+	output?: NodeJS.WriteStream;
+}
+
+/**
+ * Integrated autocomplete multiselect - combines type-ahead filtering with multiselect in one UI
+ */
+export const autocompleteMultiselect = <Value>(opts: AutocompleteMultiSelectOptions<Value>) => {
+	// Create text prompt which we'll use as foundation
+	const prompt = new AutocompletePrompt<Option<Value>>({
+		options: opts.options,
+		multiple: true,
+		filter: (search, opt) => {
+			return getFilteredOption(search, opt);
+		},
+		placeholder: opts.placeholder,
+		initialValue: opts.initialValues,
+		input: opts.input,
+		output: opts.output,
+		render() {
+			const formatOption = (option: Option<Value>, active: boolean, selectedValues: Value[]) => {
```

Comment:
you moved this into `render` so you can access `this.focusedValue` i imagine

we should just add another parameter, we shouldn't have to create this function every render

```ts
const formatOption = (option: Option<Value>, active: boolean, selectedValues: Value[], focusedValue: Value | undefined) => {
```